### ISSUE TITLE
false-positive(typescript): incorrect `as const` diagnostic

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#949]
reproduction steps:
```ts
let a = `a/b/${c}/${d}/e` as const;
```

expected: 👌 

actual: `'as const' is only allowed on literals (array, object, string, boolean) and enum members [E0291]`

**Comment by strager** [comment]
This feature was added in TypeScript 4.1. Will fix in quick-lint-js.

**Comment by vegerot** [comment]
documentation for whoever picks this up https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#template-literal-types

**Comment by leszek888** [comment]
Hi, can I get this one?
